resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
> How would we unexpectedly get a prerelease version given a specific source url?

I will leave that to @acharlieh.

>  And why do we care if the app has local directories or not?

Everything that ships with the app should reside in the _default_ folder. _local_ is where the KOs created/modified by the consumers of the app will live. 
https://answers.splunk.com/answers/37729/deployment-config-files-in-local-or-default.html